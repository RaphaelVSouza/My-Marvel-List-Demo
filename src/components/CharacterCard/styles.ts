import styled, { css } from 'styled-components'

export const Wrapper = styled.article`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 168px 1fr;
    height: 252px;
    width: 100%;
    max-width: 60rem;
    margin-bottom: ${theme.spacings.small};
  `}
`
export const Details = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
  `}
`

export const Name = styled.div`
  ${({ theme }) => css`
    font-weight: 600;
    font-size: 22px;
    color: ${theme.colors.primary};

    margin-bottom: ${theme.spacings.xsmall};
  `}
`

export const Description = styled.div`
  ${({ theme }) => css`
    display: -webkit-box;
    -webkit-line-clamp: 6;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  `}
`
