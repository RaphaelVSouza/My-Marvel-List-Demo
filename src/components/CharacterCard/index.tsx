import * as S from './styles'
import { apiImageTypes } from 'utils/apiImageTypes'

type CharacterCardProps = {
  id: string
  name: string
  description?: string
  thumbnail: {
    path: string
    extension: string
  }
}

const CharacterCard = ({
  id,
  name,
  description,
  thumbnail
}: CharacterCardProps) => (

  <S.Wrapper>
    <img
      src=
      {`${thumbnail.path}/${apiImageTypes.portrait.fantastic}.${thumbnail.extension}`}
      alt={/image_not_available/.test(thumbnail.path) ? 'not finded' : name}
    />
    <S.Details>
      <S.Name>{name}</S.Name>
      <S.Description>{description
        ? description
        : "No description found."}
      </S.Description>
    </S.Details>
  </S.Wrapper>
)

export default CharacterCard
