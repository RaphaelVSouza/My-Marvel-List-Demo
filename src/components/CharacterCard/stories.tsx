import { Story, Meta } from '@storybook/react'
import CharacterCard from '.'

export default {
  title: 'CharacterCard',
  component: CharacterCard,
} as Meta

export const Default: Story = () => (
  <CharacterCard
    id="1"
    name="Iron Man"
    description="Lorem ipsum dolor sit amet consectetur adipisicing elit. Laborum, porro alias vitae provident a ducimus tenetur! Odit molestias, vel beatae doloremque, iure, ullam quae iste magnam sequi consequuntur nihil dolor."
    thumbnail={{
      path: 'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55',
      extension: 'jpg',
    }}
  />
)
