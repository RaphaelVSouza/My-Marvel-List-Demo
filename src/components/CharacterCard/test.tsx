import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CharacterCard from '.'

const TEST_IMAGE = {
  path: 'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55',
  extension: 'jpg',
}

describe('<CharacterCard />', () => {
  it('should render the character card', () => {
    const { container } = renderWithTheme(
      <CharacterCard
        id="1"
        name="Iron Man"
        thumbnail={TEST_IMAGE}
        data-testid="character-card"
      />
    )

    expect(screen.getByRole('article')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
