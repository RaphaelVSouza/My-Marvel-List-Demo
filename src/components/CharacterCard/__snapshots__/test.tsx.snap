// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> should render the character card 1`] = `
<article
  class="sc-bdnxRM hGpXIA"
>
  <img
    alt="Iron Man"
    src="http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55/portrait_fantastic.jpg"
  />
  <div
    class="sc-gtsrHT jHJDAW"
  >
    <div
      class="sc-dlnjwi hJtaiz"
    >
      Iron Man
    </div>
    <div
      class="sc-hKFxyN guThvP"
    >
      No description found.
    </div>
  </div>
</article>
`;
